### ISSUE TITLE
Move backward/forward through the position slider (macOS)

### ISSUE THREAD

**Issue by llucps** [jonaski/strawberry#201]
Hi, 

I have created a video so it's easier to understand, but basically navigating a song using the positions slider doesn't work quite well.. many times when I click on any positions it just doesn't work.. it only works and sometime by clicking and dragging the slider as you can see in the video. I tried this on a local song in my library and from the Airsonic tab and it happens in both cases.

I also tried using a build from and my won compilation in both cases it's the same behavior.

Does it happen to you? could it be a problem with the mac version? Just asking whether it's just me or anybody else.

http://www.squeakycarrot.com/strawberry.mp4

Thanks

**Comment by jonaski** [comment]
I don't quite understand what you mean the problem is. When you click a position on the seekbar it does not immediately jump there, or does not go there at all?
Are these local files or internet streams? Or subsonic? If subsonic is the server on your local network or remote?
It's not normal for local files on your disk to behave this way.
Its normal that it takes time to for it to go to the location if they are on internet.
For me subsonic responds immediately like a local file, but I have it in a vm on this machine.
For Tidal and Qobuz it does not respond immediately when I click a place on the seekbar I need to wait 1-2 seconds before it goes to the position and starts buffering, but the position is reported by the engine (gstreamer) and the position on the UI isn't updated before it receives the position back from the engine.

**Comment by llucps** [comment]
The problem is clicking on any position on the seekbar doesn't work at all., no matter how hard you try the only way that sometimes works is by dragging the ball and not always. This happens on local files and Subsonic server I have, so in this case these are remote files. But that's not the point really... I'm thinking that is a macOS specific problem rather than general, since it seems to work for you.

So, clicking on the seekbar doesn't take you the specific position.. it does nothing.. only dragging and just sometimes work.. always forward never work going backwards.

I don't expect you to solve it since you don't have a Mac, I just wanted to know whether it was just a mac problem or it happened on any platform.

Thanks!

**Comment by alekksander** [comment]
no issues here (ubuntu based linux)
https://streamable.com/ge8ja

**Comment by llucps** [comment]
Hi,

In that case it must be a MacOS specific problem I guess.. oh well nothing I can do about it unfortunately.. Thanks anyway!

**Comment by jonaski** [comment]
@martindelille

**Comment by llucps** [comment]
Hi there,

In case it helps, I think problem lies with the tooltip that shows the ellapsed and remaining time when the cursor is on top of the slider.. it basically interferes with the click and being able to go to any point in time.. it just in the way. For some reason sometimes the tooltip doesn't show up and then you can click and it works..

This is the tooltip I'm talking about

![Screenshot 2019-07-26 at 12 33 12](https://user-images.githubusercontent.com/25823659/61945957-9cfe0780-afa1-11e9-8f2a-06375deee772.jpg)

Thanks to look into it.

**Comment by llucps** [comment]
Although I'm not a professional developer, I can read a bit of code.. so I tried to fix this issue on my own.

After a bit of research I guessed the problem was with the file /src/wdigets/tracksliderslider.cpp, I figured that the problem has to be with the popup tooltip.. so I hide the popup and it worked! I now can move the slider.. I know is a total hack but I just wanted to mention in case it helps for the person to find a proper fix.

```
void TrackSliderSlider::enterEvent(QEvent* e) {
  QSlider::enterEvent(e);
  if (isEnabled()) {
    popup_->hide();
  }
}
```

Thanks,
